module/VuFind/tests/unit-tests/src/VuFindTest/Record/LoaderTest.php

```diff
@@ -170,19 +227,49 @@ public function testBatchLoad()
         $worldCatParams = new ParamBag();
         $worldCatParams->set('fq', 'id:test4');
 
-        $factory = $this->createMock(\VuFind\RecordDriver\PluginManager::class);
+        $factory = $this->getMockBuilder(\VuFind\RecordDriver\PluginManager::class)
+            ->disableOriginalConstructor()
+            ->getMock();
         $factory->expects($this->once())->method('get')
             ->with($this->equalTo('Missing'))
             ->will($this->returnValue($missing));
 
-        $service = $this->createMock(\VuFindSearch\Service::class);
-        $service->expects($this->exactly(3))->method('retrieveBatch')
+        $commandObj = $this->getMockBuilder(\VuFindSearch\Command\AbstractBase::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $commandObj->expects($this->exactly(3))->method('getResult')
+            ->willReturnOnConsecutiveCalls($collection1, $collection2, $collection3);
+
+        $service = $this->getMockBuilder(\VuFindSearch\Service::class)
+            ->disableOriginalConstructor()->getMock();
+
+        $checkCommand1 = function ($command) {
+            return get_class($command) === \VuFindSearch\Command\RetrieveBatchCommand::class
+                && $command->getTargetIdentifier() === "Solr"
+                && $command->getArguments()[0] === ["test1", "test2"]
+                && $command->getArguments()[1]->getArrayCopy() === ['fq'=> ['id:test1']];
```

Comment:
This is another place where you might be able to compare literally against the params object instead of using getArrayCopy, since I believe we're actually checking for a specific instance and not just for a generic set of parameters.

Reply:
(See also the two other check commands below).